### PR TITLE
Upgrade laravel to 5.4

### PR DESCRIPTION
**Pull request by cschra**

### CODE REVIEW

**Comment by cschra** [comment]
@NinoRy please have a look at f8a7208 - I upgraded the Billing dependencies. The box Package is now on v. 3.0. I did the steps to upgrade from [here](https://github.com/box/spout/blob/0f0bf648024040d21c6bd9930299265e8fc42550/UPGRADE-3.0.md) but it would be nice if you could look over this.

**Comment by cschra** [comment]
@NinoRy please have a look at f8a7208 - I upgraded the Billing dependencies. The box Package is now on v. 3.0. I did the steps to upgrade from [here](https://github.com/box/spout/blob/0f0bf648024040d21c6bd9930299265e8fc42550/UPGRADE-3.0.md) but it would be nice if you could look over this.

**Comment by cschra** [comment]
@olebowle please have a look at a4cf9d5 as it changes the interface you defined in 92917a0

**Comment by cschra** [comment]
@xee8ai Please have a look at c2f590c and check if it affects anything inside the envia module

**Comment by cschra** [comment]
If everyone approves I will continue with upgrade to 5.5

File: modules/HfcCustomer/Entities/Mpr.php [hunk 1]
```diff
@@ -100,7 +99,7 @@ public function view_has_many()
      * @param 	object 	single Modem or null for all modems
      * @author: Torsten Schmidt, Nino Ryschawy
      */
-    public static function refresh($modem = null)
```
Comment (NinoRy) [inline modules/HfcCustomer/Entities/Mpr.php]: really possible?

**Pull request merged**